oci/defs.bzl

```diff
@@ -1,13 +1,77 @@
 "Public API"
 
+load("@aspect_bazel_lib//lib:jq.bzl", "jq")
 load("//oci/private:tarball.bzl", _oci_tarball = "oci_tarball")
 load("//oci/private:image.bzl", _oci_image = "oci_image")
 load("//oci/private:image_index.bzl", _oci_image_index = "oci_image_index")
 load("//oci/private:push.bzl", _oci_push = "oci_push")
 load("//oci/private:structure_test.bzl", _structure_test = "structure_test")
+load("@bazel_skylib//lib:types.bzl", "types")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
 
 oci_tarball = _oci_tarball
 oci_image = _oci_image
 oci_image_index = _oci_image_index
-oci_push = _oci_push
+oci_push_rule = _oci_push
 structure_test = _structure_test
+
+def stamped_tags(name, image_tags, **kwargs):
```

Comment:
is this going to be in our public API contract once we hit 1.0? I'd like to keep our public API lean. 
I don't have any suggestions about how to keep this userland though.

Reply:
It could go to aspect_bazel_lib perhaps. Or it can go in userland to start. I could imagine a tool chain for a expand_template binary

Reply:
maybe userland is a better start.  I have too many ideas for useful utilities like this but I don't do any of them because they don't belong to the public API.

Reply:
moved to the example